build.sh

```diff
@@ -19,17 +19,32 @@
 # * limitations under the License.
 # *
 
-#set -e # error out if required
+set -e # error out if required
 SCRIPT_EXEC="$(realpath $0)"
 MY_DIR="$(dirname $SCRIPT_EXEC)"
 
 FRAMEWORK_DIR=${MY_DIR}/framework
 UT_CONTROL_LIB_DIR=${FRAMEWORK_DIR}/ut-control
 
+# Check if the arguments contain 'TARGET=arm'
+if [[ "$*" == *"TARGET=arm"* ]]; then
```

Comment:
upper Makefile is not setting it. Ex [here](https://github.com/rdkcentral/rdk-halif-test-hdmi_cec/blob/main/Makefile). Are you suggesting they should?

Reply:
Where is it not set?

```makefile
ifeq ($(TARGET),)
$(info TARGET NOT SET )
$(info TARGET FORCED TO Linux)
TARGET=linux
SRC_DIRS += $(ROOT_DIR)/skeletons/src
YLDFLAGS += -lpthread -lrt
endif

export TARGET
```

Reply:
You are right, from makefile, it is fine. However when it gets called for the first time from [ut/build.sh](https://github.com/rdkcentral/rdk-halif-test-hdmi_cec/blob/main/build.sh) , we need to extract TARGET from there.

I tried w/o this check and found that the directory structure is getting disrupted because of this as TARGET is not set for the first time, however it doesn't fail as it gets build correctly when its called the second time from Makefile.